backend/data_handling/data_processing.py

```diff
@@ -0,0 +1,79 @@
+import pandas as pd
+from data_handling.data_maps import detailed_category_map, sum_category_map
+import random
+
+
+"""
+helper function to filter and simplify the JSON returned by the 
+[transactions/sync] endpoint into the relevant fields of
+amount, date, currency, primary, and detailed categories. 
+"""
+def clean_transaction_data(transaction_json):
+    cleaned_data = []
+    for transaction in transaction_json['transactions']:
```

Comment:
Is 'transactions' guaranteed to exist as a key in the `transaction_json`? Wondering if we need to be more defensive here to prevent possible crashes

Reply:
I would think so, but I don't know for sure how Plaid would handle the case that a user didn't use their account at all for the last cycle. I can add a simple guard here to be safe.

Reply:
I agree adding a simple guard would help protect incase Plaid ever returns an unexpected JSON. You may say "this won't happen", but the amount of times I've seen someone say that, and it happens is just way too many times